scrapers/covid-19/historical.js

```diff
@@ -77,6 +77,29 @@ const historicalV2 = async (keys, redis) => {
 			: parsedAtIndex['Province/State'].toLowerCase();
 		return newElement;
 	});
+	/* This is an edge case to handle recovered data for Canada.
+	   Since JHU dataset doesn't have recovered data for Canada
+	   disaggregated by Province/Region, the aggregate data is
+	   lost when recovered data is formatted in formatRecoveredData
+	   function. This  will create a non-existent Province/Region
+	   called recovered-aggregate with cases and deaths set to 0
+	   and recovered is set to aggregate recovered data. This data
+	   is pulled only if historical data for Canada is requested.
+	*/
+	for (let i = 0; i < result.length; i++) {
+		if (result[i].country.toLocaleLowerCase() === 'canada') {
+			const data = { country: result[i].country, countryInfo: { ...result[i].countryInfo }, province: 'recovered-aggregate', timeline: { cases: {}, deaths: {}, recovered: {} } };
+			const recoveredCanada = parsedRecovered.find(country => country['Country/Region'] === 'Canada' && country['Province/State'] === '');
+			if (!recoveredCanada) break;
+			Object.keys(data.timeline.recovered).forEach(date => {
+				data.timeline.recovered[date] = parseInt(recoveredCanada[date]);
+				data.timeline.deaths[date] = 0;
+				data.timeline.cases[date] = 0;
+			});
+			result.splice(i, 0, data);
```

Comment:
What does this do?

Reply:
Essentially I am inserting a fake `Region/Province` for Canada in the dataset which will hold cummulative aggregate data. All other `Regions/Provinces` have  `deaths`, `cases` and `recovered` fields in the `timeline`.  I am doing the same for the fake `Region/Province` I am inserting. The only difference is that  this fake `Region/Province` is supposed to hold aggregate recovered data for all provinces of Canada. I have set the `deaths` and `recovered` fields to 0 since that data will be extracted from the dataset once requested.